jwql/utils/engineering_database.py

```diff
@@ -0,0 +1,174 @@
+        self.start_time = start_time
+        self.end_time = end_time
+        self.data = data
+        self.meta = meta
+        self.info = info
+
+    def __str__(self):
+        """Return string describing the instance."""
+        return 'EdbMnemonic {} with {} records between {} and {}'.format(
+            self.mnemonic_identifier, len(self.data), self.start_time.isot,
+            self.end_time.isot)
+
+    # def interpolate(self, date, **kwargs):
+    #     """Interpolate value at times specified in data argument."""
+    #     raise NotImplementedError
+
+
+def process_mast_service_request_result(result):
+    """Parse the result of a MAST EDB query into a astropy data table and a meta dictionary.
+
+    Parameters
+    ----------
+    result : list of requests.models.Response instances
+        The object returned by a call to Mast.service_request_async
+
+    Returns
+    -------
+        data : astropy.table.Table
+            Table representation of the returned data.
+        meta : dict
+            Additional information returned by the query
+
+    """
+    json_data = result[0].json()
+    if json_data['status'] != 'COMPLETE':
+        print(json_data['status'])
+        raise RuntimeError('mnemonic query did not complete.')
+
+    try:
+        # timestamp-value pairs in the form of an astropy table
+        data = Table(json_data['data'])
+    except KeyError:
+        raise RuntimeError('Query did not return any data.')
+
+    # collect meta data
+    meta = {}
+    for key in json_data.keys():
+        if key.lower() != 'data':
+            meta[key] = json_data[key]
+
+    return data, meta
+
+
+def get_all_mnemonic_identifiers():
+    """Return identifiers and meta data for all mnemonics in the DMS engineering database."""
+    out = Mast.service_request_async(mast_edb_mnemonic_service, {})
+    data, meta = process_mast_service_request_result(out)
+    return data, meta
+
+
+def query_mnemonic_info(mnemonic_identifier):
+    """Query the EDB to return the mnemonic description.
+
+    Parameters
+    ----------
+    mnemonic_identifier: str
+        Telemetry mnemonic identifier, e.g. 'SA_ZFGOUTFOV'
+
+    Returns
+    -------
+        info : dict
+            Object that contains the returned data
+
+    """
+    parameters = {"mnemonic": "{}".format(mnemonic_identifier)}
+    result = Mast.service_request_async(mast_edb_dictionary_service, parameters)
+    info = process_mast_service_request_result(result)[0]
+    return info
+
+
+def query_single_mnemonic(mnemonic_identifier, start_time, end_time):
+    """Query the DMS EDB to return the mnemonic readings between start_time and end_time.
+
+    Parameters
+    ----------
+    mnemonic_identifier : str
+        Telemetry mnemonic identifier, e.g. 'SA_ZFGOUTFOV'
+    start_time : astropy.time.Time instance
+        Start time
+    end_time : astropy.time.Time instance
+        End time
+
+    Returns
+    -------
+        mnemonic : instance of EdbMnemonic
```

Comment:
@Johannes-Sahlmann Oops, I think this accidentally got indented.

Reply:
Fixed (there were other instances.) Again, do you have an automated way to detect these?